state.go

```diff
@@ -281,7 +287,12 @@ func (m *Memberlist) probeNode(node *nodeState) {
 		var msgs [][]byte
 		if buf, err := encode(pingMsg, &ping); err != nil {
 			m.logger.Printf("[ERR] memberlist: Failed to encode ping message: %s", err)
-			return
+			switch m.transport.ProtocolType() {
+			case ProtocolTypeTCP:
+				goto TCPCONTINUE
+			default:
+				return
+			}
```

Comment:
When using a TCP protocol, this used to return instantly when the other end was not available. That prevent this node from being marked as suspect and failed.